Common/TableProducer/occupancyTableProducer.cxx

```diff
@@ -0,0 +1,1053 @@
+      }
+      return;
+    }
+
+    const int nBCinTF = 114048;         /// CCDB
+    const int nBCinDrift = 114048 / 32; /// to get from ccdb in future
+
+    // Occupancy Maps Per DF // a DF contains one or more timesFrames (TFs)
+    std::unordered_map<int64_t, std::vector<int64_t>> BC_TF_Map;
+
+    std::unordered_map<int64_t, std::vector<float>> occ_Prim_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_FV0A_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_FV0C_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_FT0A_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_FT0C_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_FDDA_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_FDDC_Unfm_80;
+
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrack_PVC_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrack_ITS_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrack_TPC_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrack_TRD_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrack_TOF_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrackSize_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrackTPC_A_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrackTPC_C_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrackITS_TPC_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrackITS_TPC_A_Unfm_80;
+    std::unordered_map<int64_t, std::vector<float>> occ_NTrackITS_TPC_C_Unfm_80;
+    //
+    // Calculation of TFid and BCinTF
+    auto bc = collisions.iteratorAt(0).bc_as<aod::BCsWithTimestamps>();
+    int run = bc.runNumber();
+    auto time = bc.timestamp();
+
+    auto runDuration = ccdb->getRunDuration(run, true);
+    int64_t tsSOR = runDuration.first;
+    auto ctpx = ccdb->getForTimeStamp<std::vector<Long64_t>>("CTP/Calib/OrbitReset", tsSOR);
+    int64_t tsOrbitReset = (*ctpx)[0];
+    uint32_t nOrbitsPerTF = run < 534133 ? 128 : 32;
+    int64_t orbitSOR = (tsSOR * 1000 - tsOrbitReset) / o2::constants::lhc::LHCOrbitMUS;
+    orbitSOR = orbitSOR / nOrbitsPerTF * nOrbitsPerTF;
+    int64_t bcSOR = orbitSOR * nBCsPerOrbit + customOrbitOffset * nBCsPerOrbit;
+    int32_t nBCsPerTF = nOrbitsPerTF * nBCsPerOrbit;
+    // nBCsPerITSROF = (run >= 543437 && run <= 545367) ? 594 : 198;
+
+    int64_t TFidThis = (bc.globalBC() - bcSOR) / nBCsPerTF;
+    int bcInTF = (bc.globalBC() - bcSOR) % nBCsPerTF;
+    //
+    int iColl = -1;
+    int lastRun = -1;
+    std::vector<int64_t> TFIDList;
+
+    for (const auto& collision : collisions) {
+      iColl++;
+
+      bc = collision.bc_as<aod::BCsWithTimestamps>();
+      run = bc.runNumber();
+      if (run != lastRun) {
+        // lastRun = run;
```

Comment:
GetRunInfo and GetTimingInfo are used now, runNumber, even though it wont change for dataframes, it will change when we will execute in on mutltiple runs